### PR TITLE
Initial spec and minimal parser

### PR DESCRIPTION
**Pull request by robsimmons**

### CODE REVIEW

File: src/ast.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+export interface Position {
+    readonly line: number;
+    readonly column: number;
+}
+
+export interface SourceLocation {
+    readonly start: Position;
+    readonly end: Position;
+    readonly source?: string | null;
```
Comment (chrisamaphone) [inline src/ast.ts]: does this need to be string | null if it's already optional?

File: syntax/proposition.ne [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
+
+@{%
+const lexer = require('./lex').lexer;
+const util = require('./parse/parse');
+%}
+
+@lexer lexer
+
+Proposition -> _ Prop _   {% x => x[1] %}
```
Comment (chrisamaphone) [inline syntax/proposition.ne]: indent with everything else

**Review by chrisamaphone — APPROVED** [review]

File: src/ast.ts (hunk 1 above)
Comment (robsimmons) [reply src/ast.ts]: The ESprima spec, which I generally follow, does use the optional `string | null`, but I don't see a reason not to change this to a mandatory `string | null`.

**Comment by robsimmons** [comment]
feat: basic parsing infrastructure in place

**Pull request merged**